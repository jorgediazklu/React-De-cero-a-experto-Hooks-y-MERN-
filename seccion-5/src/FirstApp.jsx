import React from "react";
import PropTypes from "prop-types";

export const FirstApp = ({ title, subTitle, name }) => {
  return (
    <div>
      <h1>{title}</h1>
      <p>{subTitle}</p>
      <p>{name}</p>
    </div>
  );
};
FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
};
FirstApp.defaultProps = {
  name: "No hay nombre",
  //   title: "No hay titulo",
  subTitle: "No hay subtitulo",
};
