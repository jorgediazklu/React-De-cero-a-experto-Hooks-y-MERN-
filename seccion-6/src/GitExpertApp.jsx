import React, { useState } from "react";
import { AddCategories, GifGrid } from "./components";

export const GitExpertApp = () => {
  const [categories, setCategories] = useState(["One Punch"]);
  const onAddCategory = (inputValue) => {
    if (categories.includes(inputValue)) return;
    setCategories((prev) => [...prev, inputValue]);
  };
  return (
    <div>
      GitExpertApp
      <br />
      <br />
      <br />
      <AddCategories onAddCategory={onAddCategory} />
      {categories.map((category) => (
        <GifGrid key={category} category={category} />
      ))}
    </div>
  );
};
