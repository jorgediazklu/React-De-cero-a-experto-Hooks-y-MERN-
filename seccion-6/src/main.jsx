import React from "react";
import ReactDOM from "react-dom/client";
import { GitExpertApp } from "./GitExpertApp";
import "./styles.css";
// pGyq9JNgrL4MkrvIY2TUBlpYOglFFs17
ReactDOM.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    <GitExpertApp />
  </React.StrictMode>
);
