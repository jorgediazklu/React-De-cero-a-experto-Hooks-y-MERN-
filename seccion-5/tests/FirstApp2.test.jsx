import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe("prueba a <FirstApp/>", () => {
  const title = "title";

  test("match con snapshot", () => {
    const { container } = render(<FirstApp title={title} />);
    expect(container).toMatchSnapshot();
  });
  test("mostar el title", () => {
    render(<FirstApp title={title} />);
    //  el screen es el html generado por el render
    //  not: negacion
    //  expect(screen.getByText(title)).not.toBeTruthy()
    expect(screen.getByText(title)).toBeTruthy();
  });
  test("mostar el title en h1", () => {
    render(<FirstApp title={title} />);
    //getByRole: obtiene las etiqutas html
    //heading: obtiene la etiqueta h1,h2,h3,h4,h5,h6
    //level: obtiene la primera etiqueta h1 que encuentre
    //innerHtml: lo que esta escrito
    //toContain: que contenga el title
    expect(screen.getByRole("heading", { level: 1 }).innerHTML).toContain(
      title
    );
  });
});
