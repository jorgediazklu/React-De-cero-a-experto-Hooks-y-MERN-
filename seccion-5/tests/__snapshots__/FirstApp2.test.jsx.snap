// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prueba a <FirstApp/> match con snapshot 1`] = `
<div>
  <div>
    <h1>
      title
    </h1>
    <p>
      No hay subtitulo
    </p>
    <p>
      No hay nombre
    </p>
  </div>
</div>
`;

exports[`prueba a <FirstApp/> match snapshot 1`] = `
<div>
  <div>
    <h1>
      titulo
    </h1>
    <p>
      No hay subtitulo
    </p>
    <p>
      No hay nombre
    </p>
  </div>
</div>
`;
